File: bubble/cpp/plot_output.m
A = readtable("output/sim_data_2026-05-03_12-13-07.csv");
sim = table2array(A);
B = readtable("output/sim_data_2026-05-03_12-13-07.csv");
split = table2array(B);

x0 = sim(:,2) ./ 1e5;
xf = sim(:,3) ./ 1e5;
xs = split(:,4) ./ 1e5;
En0 = sim(:,4); 
En1 = sim(:,5);
Ens = split(:,6);

%%
% bins = logspace(0,1,10);
% n0 = histcounts(En0,bins);
% n1 = histcounts(En1,bins);
% ns = histcounts(Ens,bins);
% n0(end+1) = 0.0; n1(end+1) = 0.0; ns(end+1) = 0.0;
% 
% figure()
% hold on
% grid on
% scatter(bins,n0./sum(n0),'filled')
% scatter(bins,n1./sum(n1),'filled')
% scatter(bins,ns./sum(ns),'filled')
% set(gca,'XScale','log','YScale','log');

[n0,bins0] = histcounts(En0,25);
[n1,bins1] = histcounts(Enf,25);
n0(end+1) = 0.0; n1(end+1) = 0.0;
figure()
hold on
grid on
scatter(bins0,n0./sum(n0))
scatter(bins1,n1./sum(n1))
set(gca,'XScale','log','YScale','log');

figure()
hold on
grid on
scatter(x0,En0,'filled')
scatter(xf,En1,'filled')
scatter(xs,Ens,'filled')
set(gca,'YScale','log');

File: bubble/cpp/plot_output.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Load the data
# Using pandas to read the CSV (equivalent to readtable)
A = pd.read_csv("output/sim_data_2026-05-03_12-13-07.csv")
sim = A.values # equivalent to table2array
B = pd.read_csv("output/sim_data_2026-05-03_12-13-07.csv")
split = B.values

# Extract and scale columns (Note: Python uses 0-based indexing)
x0 = sim[:, 1] / 1e5
xf = sim[:, 2] / 1e5
xs = split[:, 3] / 1e5
En0 = sim[:, 3]
En1 = sim[:, 4]
# Ens = split[:, 5]

# Histogram Figure
plt.figure()
# MATLAB's histcounts(data, 25) returns 25 bins (26 edges)
n0, bins0 = np.histogram(En0, bins=25)
n1, bins1 = np.histogram(En1, bins=25)

# Normalizing and plotting (scatter uses the left edge of bins here)
# Added 0.0 to match MATLAB's manual padding if needed for alignment
plt.scatter(bins0[:-1], n0 / np.sum(n0), label='En0', facecolors='none', edgecolors='b')
plt.scatter(bins1[:-1], n1 / np.sum(n1), label='En1', facecolors='none', edgecolors='r')

plt.xscale('log')
plt.yscale('log')
plt.grid(True)
plt.legend()
plt.title("Normalized Histograms")

# Scatter Plot Figure
plt.figure()
plt.scatter(x0, En0, label='Initial', alpha=0.6)
plt.scatter(xf, En1, label='Final', alpha=0.6)
# plt.scatter(xs, Ens, label='Split', alpha=0.6)

plt.yscale('log')
plt.grid(True)
plt.legend()
plt.xlabel('Position (scaled)')
plt.ylabel('Energy')

plt.show()
